filament/src/details/View.cpp

```diff
@@ -973,19 +973,38 @@ void FView::executePickingQueries(backend::DriverApi& driver,
         // adjust for dynamic resolution and structure buffer scale
         const uint32_t x = uint32_t(float(pQuery->x) * (scale * mScale.x));
         const uint32_t y = uint32_t(float(pQuery->y) * (scale * mScale.y));
-        driver.readPixels(handle, x, y, 1, 1, {
-                &pQuery->result.renderable, 4u * 4u, // 4*float
-                backend::PixelDataFormat::RG, backend::PixelDataType::FLOAT,
-                pQuery->handler, [](void*, size_t, void* user) {
-                    FPickingQuery* pQuery = static_cast<FPickingQuery*>(user);
-                    float const identity = *((float*)((char*)&pQuery->result.renderable));
-                    pQuery->result.renderable = Entity::import(identity);
-                    pQuery->result.fragCoords = {
-                            pQuery->x, pQuery->y, float(1.0 - pQuery->result.depth) };
-                    pQuery->callback(pQuery->result, pQuery);
-                    FPickingQuery::put(pQuery);
-                }, pQuery
-        });
+
+        if (UTILS_UNLIKELY(driver.getFeatureLevel() == FeatureLevel::FEATURE_LEVEL_0)) {
+            driver.readPixels(handle, x, y, 1, 1, {
+                    &pQuery->result.reserved1, 4u, // 4
+                    backend::PixelDataFormat::RGBA, backend::PixelDataType::UBYTE,
+                    pQuery->handler, [](void*, size_t, void* user) {
+                        FPickingQuery* pQuery = static_cast<FPickingQuery*>(user);
+                        int32_t const identity = int32_t(pQuery->result.reserved1 >> 8);
+                        float const depth = float(pQuery->result.reserved1 & 0xFF) / 255.0f;
```

Comment:
Looks like the alpha bit is the depth, but in the shader it reads like the r channel is the depth?

Reply:
no because actually RGBA8 / UBYTE ends up being R,G,B,A in memory which is 0xABGR when loaded as a uint32_t. But this is a good point though that I'm relying on little-endian. I should fix/clarify that.

Reply:
made it endian proof